### PR TITLE
[Improve] Add @Nullable annotation to BaseFlinkSink apis

### PR DESCRIPTION
**Pull request by asdf2014**
## Purpose of this pull request

We should add `@Nullable` annotation to `BaseFlinkSink` apis to remind returns of them could be `null`.

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connector-flink-doris/src/main/java/org/apache/seatunnel/flink/sink/DorisSink.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import org.apache.flink.types.Row;
 import org.apache.flink.util.Preconditions;
 
+import javax.annotation.Nullable;
```
Comment (SteNicholas) [inline seatunnel-connectors/seatunnel-connector-flink-doris/src/main/java/org/apache/seatunnel/flink/sink/DorisSink.java:39]: Does the `javax` package class import after the `java`?

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-doris/src/main/java/org/apache/seatunnel/flink/sink/DorisSink.java:37]: Hi @SteNicholas , I believe it’s not big deal, also you are welcome to create a PR to add a rule into code style if you are free

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-doris/src/main/java/org/apache/seatunnel/flink/sink/DorisSink.java:39]: Hi @SteNicholas, thanks for you comment, I agree with you, and I just create a PR to add a rule into Code Style for the order of improts to further ensure uniformity of code style across each connector.

**Comment by SteNicholas** [comment]
@asdf2014 , could the `@Nullable` annotation follow the checkstyle of Flink: https://flink.apache.org/contributing/code-style-and-quality-java.html
- Use @Nullable annotation where you do not use Optional for the nullable values.
- If you can prove that Optional usage would lead to a performance degradation in critical code then fallback to @Nullable.
- Always use Optional to return nullable values in the API/public methods except the case of a proven performance concern.
- Do not use Optional as a function argument, instead either overload the method or use the Builder pattern for the set of function arguments.
Note: an Optional argument can be allowed in a private helper method if you believe that it simplifies the code (example).
- Do not use Optional for class fields.

**Comment by asdf2014** [comment]
Hi @SteNicholas, thanks for your comment. This is great, it would be better to add them as rules into check style plugin, right? Would you willing to raise up some follow-up PRs to implement them

**Comment by SteNicholas** [comment]
@asdf2014, it's recommended to add them as rules into check style plugin. I'm willing to raise up some follow-up PRs to implement them.

File: seatunnel-connectors/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/sink/KafkaTable.java [hunk 2]
```diff
@@ -38,6 +38,8 @@
 import org.apache.flink.table.descriptors.Schema;
 import org.apache.flink.types.Row;
 
+import javax.annotation.Nullable;
+
 import java.util.Properties;
```
Comment (SteNicholas) [inline seatunnel-connectors/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/sink/KafkaTable.java:43]: Does the checkstyle rules for `javax` work?

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/sink/KafkaTable.java:43]: Yes, it does

**Review by SteNicholas — DISMISSED** [review]
LGTM. @CalvinKirs could you review this pull request?

**Comment by CalvinKirs** [comment]
> LGTM. @CalvinKirs could you review this pull request?

Sorry for taking so long to reply, thanks for your review, I think good

**Comment by CalvinKirs** [comment]
@asdf2014 hi, can you resolve conflicts files?

**Comment by asdf2014** [comment]
Hi @CalvinKirs, thanks for your comment, the conflicts have been resolved

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**